ipn/ipnlocal/local.go

```diff
@@ -342,7 +342,17 @@ func (b *LocalBackend) onHealthChange(sys health.Subsystem, err error) {
 // can no longer be used after Shutdown returns.
 func (b *LocalBackend) Shutdown() {
 	b.mu.Lock()
+	if b.shutdownCalled {
+		b.mu.Unlock()
+		return
+	}
 	b.shutdownCalled = true
+
+	if b.loginFlags&controlclient.LoginEphemeral != 0 {
+		b.mu.Unlock()
+		b.logout(b.ctx, true) // best effort
+		b.mu.Lock()
```

Comment:
any invariants you need to re-check now that you've dropped the lock and reacquired it?

Reply:
Nothing can restart the backend, so no.